### PR TITLE
feat(j-s): Appeal Counter Party Email

### PR DESCRIPTION
**Pull request by gudjong**
# Appeal Counter Party Email

[Landsréttur - senda kærutilkynningu þegar verjandi kærir](https://app.asana.com/0/1199153462262248/1204367224389681/f)

## What

- Sends full user details with queued messages to support defender initiated messages.
- Sends appeal email to the counter party (defender or prosecutor) when a case is appealed.

## Why

- To keep the counter party informed of the appeal.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCaseToPolice.spec.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ describe('InternalCaseController - Deliver case to police', () => {
       const then = {} as Then
 
       await internalCaseController
-        .deliverCaseToPolice(caseId, user, theCase, { userId })
+        .deliverCaseToPolice(caseId, theCase, { user })
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCaseToPolice.spec.ts:44]: Does the user here need to be wrapped in `{}`? It's already an object, am I misunderstanding something?

File: apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCourtRecordToCourt.spec.ts [hunk 2]
```diff
@@ -44,7 +45,7 @@ describe('InternalCaseController - Deliver court record to court', () => {
       const then = {} as Then
 
       await internalCaseController
-        .deliverCourtRecordToCourt(caseId, user, theCase, { userId })
+        .deliverCourtRecordToCourt(caseId, theCase, { user })
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCourtRecordToCourt.spec.ts:48]: Same comment as ☝️

File: apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCaseToPolice.spec.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCaseToPolice.spec.ts:44]: The deliver DTO contains one field, user, which happens to be an object, so this is correct. We could let the DTO list all the fields of the user object but that would be a bit strange and would require us to change the DTO every time we change the User type.

File: apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCourtRecordToCourt.spec.ts (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/deliverCourtRecordToCourt.spec.ts:48]: Same response.

**Review by oddsson — APPROVED** [review]

**Pull request merged**